### PR TITLE
support bpmn executable property

### PR DESCRIPTION
**Pull request by burnettk**
If the bpmn executable property is not true, avoid giving the user Start buttons in the UI
also avoid start buttons if a primary file is not set
affects process model show, process group show, and the Start new process tab on the homepage

addresses #742 

## Summary by CodeRabbit

- **New Features**
	- Introduced an `is_executable` property to process models, enhancing control over process execution capabilities.
	- Added user-specific process model retrieval, improving personalized user experience.
	- Implemented conditional rendering of the start button in the UI based on process executability and the presence of a primary file.

- **Bug Fixes**
	- Refined process model display logic to include file references and executable status, ensuring accurate process model information.

- **Tests**
	- Added tests to validate new process model API functionalities and executable status handling.

### CODE REVIEW

File: spiffworkflow-backend/src/spiffworkflow_backend/routes/process_models_controller.py [hunk 1]
```diff
@@ -165,10 +165,14 @@ def process_model_show(modified_process_model_identifier: str, include_file_refe
...
+    ref_for_primary_file = None
     if include_file_references:
         for file in process_model.files:
-            file.references = SpecFileService.get_references_for_file(file, process_model)
-
+            refs = SpecFileService.get_references_for_file(file, process_model)
+            file.references = refs
+            if file.name == process_model.primary_file_name:
+                ref_for_primary_file = ProcessModelService.reference_for_primary_file(refs, process_model.primary_file_name)
+    process_model.is_executable = True if ref_for_primary_file and ref_for_primary_file.properties["is_executable"] else False
```
[the remark this answers is not in the document]
Comment (jbirddog) [reply spiffworkflow-backend/src/spiffworkflow_backend/routes/process_models_controller.py]: `properties.get("is_executable", False)` would also work and not require the logic for the key not being present.

**Review by jbirddog — APPROVED** [review]

File: spiffworkflow-backend/src/spiffworkflow_backend/routes/process_models_controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (burnettk) [reply spiffworkflow-backend/src/spiffworkflow_backend/routes/process_models_controller.py]: thanks. i had already rewritten this to use the "embellish" method that was used elsewhere. the github PR UI is pretty confusing for commented code snippets that are outdated. hard problem, i suppose.

File: spiffworkflow-backend/tests/spiffworkflow_backend/integration/test_process_models_controller.py [hunk 2]
```diff
@@ -60,3 +61,39 @@ def test_cannot_save_process_model_file_with_called_elements_user_does_not_have_
...
+            "non_executable/non_executable",
+            bpmn_file_name="non_executable.bpmn",
+            process_model_source_directory="non_executable",
+        )
+        json = self._get_process_show_show_response(client, user, process_model_1.modified_process_model_identifier())
+        assert json["is_executable"] is True
+
+        json = self._get_process_show_show_response(client, user, process_model_2.modified_process_model_identifier())
+        assert json["is_executable"] is False
+
```
[the remark this answers is not in the document]
Comment (burnettk) [reply spiffworkflow-backend/tests/spiffworkflow_backend/integration/test_process_models_controller.py]: addressed

[the remark this answers is not in the document]
Comment (burnettk) [reply spiffworkflow-backend/tests/spiffworkflow_backend/integration/test_process_models_controller.py]: it's using the fixture, it's checking an additional property to make sure it's fetching the right process model, and there is a negative test case now that confirms correct handling of model not found.

**Pull request merged**